### ISSUE TITLE
Dash in resource name: Mangling of resource names in add_resource() breaks compatibility

### ISSUE THREAD

**Issue by bhawk** [allisson/python-simple-rest-client#32]
Problem:
 recently there was resource name mangling added to add_resource() method which blocks usingn this lib for third-party API with resource names with hyphens.

Example:

Valid Resource URI: 
http://some-rest-api/resource-name

what lib does:

```
from simple_rest_client.api import API
api = API('http://some-rest-api/')
api.add_resource(resource_name='resource-name')
api.get_resource_list()
# ['resource_name']
api.resource_name.list()
# Error on GET for url 'http://some-rest-api/resource_name'
```

I understand why it was done,
but this completely breaks compatibility of this lib with any ext API which uses such resource names.

So maybe this mechanics should be converted into an additional helper - add "sanitized" attribute only if there is no such already?

**Comment by bhawk** [comment]
Ah, looks like this is https://github.com/allisson/python-simple-rest-client/issues/14

**Comment by leorochael** [comment]
Actually, this looks like the reverse of #14.

In PR #28, it seems the resource name mangling was added to attributes to match resources with hyphens in the name so they can become valid Python attributes.

But in #29, the name mangling was extended into the `Resource` name itself.

I just added a [late comment](https://github.com/allisson/python-simple-rest-client/pull/29#pullrequestreview-826994613) to PR #29 specifying what would need to change.

I'd be happy to provide a PR with my suggestions.
